test/optim/test_optim.py

```diff
@@ -834,6 +834,40 @@ def test_rprop(self):
             )
 
 
+    def test_lbfgs_complex(self):
+        for linesearch in ("strong_wolfe", None):
+            complex_params = [torch.randn(10, 5, dtype=torch.complex64, requires_grad=True), torch.randn(10, dtype=torch.complex64, requires_grad=True), torch.randn(10, 5, dtype=torch.float32, requires_grad=True)]
+            real_params = [torch.view_as_real(param).detach().clone().requires_grad_() if param.is_complex() else param.detach().clone().requires_grad_() for param in complex_params]
+            complex_steps = []
+            real_steps = []
+            grads_losses = []
+
+            def real_closure():
+                for p in real_params:
+                    p.grad = torch.randn_like(p)
+                    grads_losses.append(p.grad.clone())
+                    real_steps.append(p.clone())
+                grads_losses.append(torch.randn(1))
+                return grads_losses[-1].clone()
+
+            def complex_closure():
+                for p in complex_params:
+                    if torch.is_complex(p):
+                        p.grad = torch.view_as_complex(grads_losses.pop(0))
+                        complex_steps.append(torch.view_as_real_copy(p))
+                    else:
+                        p.grad = grads_losses.pop(0)
+                        complex_steps.append(p.clone())
+                return grads_losses.pop(0)
+
+            real_optim = LBFGS(real_params, line_search_fn=linesearch)
+            complex_optim = LBFGS(complex_params, line_search_fn=linesearch)
+            real_optim.step(real_closure)
+            complex_optim.step(complex_closure)
+            self.assertEqual(complex_steps, real_steps)
```

Comment:
```suggestion
            self.assertEqual(complex_params, real_params)
```
Why not just this?

Also, could we step multiple times? Like 3?